src/utils.js

```diff
@@ -0,0 +1,1317 @@
+  if (deploymentPath) {
+    if (projectName !== '' && projectName !== deploymentProjectName) {
+      throw new Error('The project name in the deployment file does not match the project name in the manifest file')
+    }
+  }
+
+  const filecomponents = {
+    packages: packages,
+    deploymentTriggers: deploymentTriggers,
+    deploymentPackages: deploymentPackages,
+    manifestPath: manifestPath,
+    manifestContent: manifest,
+    projectName: projectName
+  }
+  return filecomponents
+}
+
+/**
+ * Handle Adobe auth action dependency
+ *
+ * This is a temporary solution and needs to be removed when headless apps will be able to
+ * validate against app-registry
+ *
+ * This function stores the IMS organization id in the Adobe I/O cloud state library which
+ * is required by the headless validator.
+ *
+ * The IMS org id must be stored beforehand in `@adobe/aio-lib-core-config` under the
+ * `'project.org.ims_org_id'` key. TODO: pass in imsOrgId
+ *
+ * @param {object} actions the actions
+ * @param {object} owOptions OpenWhisk actions
+ * @param {string} imsOrgId the IMS Org Id
+ */
+async function setupAdobeAuth (actions, owOptions, imsOrgId) {
+  // do not modify those
+  const ADOBE_HEADLESS_AUTH_ACTION = '/adobeio/shared-validators-v1/headless'
+  const AIO_STATE_KEY = '__aio'
+  const AIO_STATE_PUT_ENDPOINT = 'https://adobeio.adobeioruntime.net/api/v1/web/state/put'
+
+  const hasAnAdobeHeadlessAuthSequence = actions.some(a => a.exec && a.exec.kind === 'sequence' && a.exec.components.includes(ADOBE_HEADLESS_AUTH_ACTION))
+  if (hasAnAdobeHeadlessAuthSequence) {
+    // if we use the headless (default auth action) we need to store the ims org id in the
+    // cloud state lib. This is needed by the auth action to perform an org check.
+    if (!imsOrgId) {
+      throw new Error('imsOrgId must be defined when using the Adobe headless auth validator')
+    }
+    const res = await fetch(AIO_STATE_PUT_ENDPOINT, {
+      method: 'post',
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: `Basic ${Buffer.from(owOptions.apiKey).toString('base64')}`
+      },
+      body: JSON.stringify({
+        namespace: owOptions.namespace,
+        key: AIO_STATE_KEY,
+        value: { project: { org: { ims_org_id: imsOrgId } } },
+        ttl: -1 // unlimited
+      })
+    })
+    if (!res.ok) {
+      throw new Error(`failed setting ims_org_id=${imsOrgId} into state lib, received status=${res.status}, please make sure your runtime credentials are correct`)
+    }
+    logger.debug(`set IMS org id into cloud state, response: ${JSON.stringify(await res.json())}`)
+  }
+}
+
+/**
+ * Deploy a package
+ *
+ * @param {object} entities the entities
+ * @param {object} ow the OpenWhisk object
+ * @param {object} logger the logger
+ * @param {string} imsOrgId the IMS Org ID
+ */
+async function deployPackage (entities, ow, logger, imsOrgId) {
```

Comment:
Does it make sense that deployPackage **must** receive an imsOrgId? This only seems to matter if adobe-auth is being used.  This needs to be used by the runtime plugin also.
attn: @shazron @moritzraho @Himavanth

Reply:
We could instead read the config only and set it into the state-lib from the app-scripts (later in app plugin), however this will come at one drawback which is that if a user deploys a manifest with the require-adobe-auth flag via the runtime plugin, the validation won't work unless the user manually sets the ims org id in his state lib

Reply:
However, this need is temporary anyways, let me explain:

- currently, actions in headless apps are deployed with the headless validator that needs the ims-org id to be stored in state lib to do an org check. This is a temporary requirement as in a few months we will be able to check JWT tokens for headless apps against the app registry validator

- the current implementation in the runtime plugin helpers: https://github.com/adobe/aio-cli-plugin-runtime/blob/master/src/runtime-helpers.js#L842 reads the config directly instead of taking the ims org id as input. 

- I would suggest to leave it like it is in the runtime helpers, I understand that we shouldn't read the config from the aio-lib-runtime, but that would be the easiest for the time being, as we would remove this in a few months anyways

Reply:
app-scripts is not called by runtime-plugin
My ask is that we assume the action does NOT require adobe-auth and only require imsOrgId IF the sequence includes _secure_ action/sequence

Reply:
we WILL read aio config from runtime-plugin, however I would expect in many cases it will be undefined

Reply:
> app-scripts is not called by runtime-plugin
sry if my response was not clear, I did not meant that

> My ask is that we assume the action does NOT require adobe-auth and only require imsOrgId IF the sequence includes secure action/sequence

the imsOrgId is required IF the `require-adobe-auth` annotation is set and the app is headless

> we WILL read aio config from runtime-plugin, however I would expect in many cases it will be undefined

ok then we should make sure to read it from `rt sync`, `rt deploy` and the app-scripts `deploy` command.

Reply:
> app-scripts is not called by runtime-plugin

sry for the misunderstanding but I did not meant that app-scripts is called by rt-plugin, I thought you suggested to only support the adobe-auth in app-scripts/app-plugin

 > My ask is that we assume the action does NOT require adobe-auth and only require imsOrgId IF the sequence includes secure action/sequence

I am not sure to understand, if `require-adobe-auth` is set (and the app is headless) then `imsOrgId` is required. Also we create the secure sequence in processPackage IF `require-adobe-auth` is set